forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/api/authorisation/redirect/AuthorisationApiController.java

```diff
@@ -270,16 +272,61 @@ private SignedJWT validateRequestParameter(String responseType, String clientId,
             verifyQueryParameterMatchesRequestParameterClaim(requestParameters, "client_id", clientId);
             Optional<Tpp> byClientId = tppStoreService.findByClientId(clientId);
             if (!byClientId.isPresent()) {
-                throw new OBErrorException(OBRIErrorType.REQUEST_PARAMETER_JWT_FORMAT_INVALID, "Unknown client id '" + clientId + "'");
+                throw new OBErrorException(OBRIErrorType.REQUEST_PARAMETER_JWT_FORMAT_INVALID, "Unknown client id '"
+                        + clientId + "'");
             }
             Tpp tpp = byClientId.get();
 
-            log.debug("Validate the request parameter");
-            if (tpp.getRegistrationResponse().getJwks() != null) {
-                cryptoApiClient.validateJwsWithJWK(requestParametersSerialised, clientId, tpp.getRegistrationResponse().getJwks().getKeys().get(0).toJSONString());
+            log.debug("Validate the request parameter signature");
+            SignedJWT validatedJwt = null;
+            boolean validated = false;
+            OIDCRegistrationResponse registrationResponse = tpp.getRegistrationResponse();
+            if (registrationResponse != null) {
+                JWKSet jwkSet = registrationResponse.getJwks();
+                if (jwkSet != null) {
+                    List<JWK> jwkSetKeys = jwkSet.getKeys();
+                    if (jwkSetKeys != null && !jwkSetKeys.isEmpty()) {
+                        JWK jwk = jwkSetKeys.get(0);
+                        String jwksKeys = jwk.toString();
+                        log.debug("validateRequestParameters() tpp has jwksKeys as part of registraiton. They will be" +
+                                " used to validate the request parameter.");
+                        // ToDo: This throws is the jwt can't be validated. I was checking to see if the return (a
+                        //  SignedJWT was non-null as well, but it would appear that the API can return a null
+                        //  SignedJWT even when the jwt is valid. This seems wrong but I have to stop fixing stuff
+                        //  somewhere or I will never get this PR merged!!!
+                        cryptoApiClient.validateJwsWithJWK(requestParametersSerialised, clientId, jwksKeys);
+                        validated = true;
+                    } else {
+                        log.error("validateRequestParameter() tpp has no jwkSetKeys; {}", tpp);
```

Comment:
If this is genuinely an error, should we be throwing an exception here, or is warning more appropriate?

Reply:
I think you're not throwing an error, because you want to try `validateJws()` below, but I would still be tempted to make this a warning.

Reply:
Fixed